troposphere/static/js/utilities/subscribe.js

```diff
@@ -0,0 +1,97 @@
+import ResourceRequestStore from "stores/ResourceRequestStore";
+import IdentityMembershipStore from "stores/IdentityMembershipStore";
+import StatusStore from "stores/StatusStore";
+import SSHKeyStore from "stores/SSHKeyStore";
+import QuotaStore from "stores/QuotaStore";
+import SizeStore from "stores/SizeStore";
+import TagStore from "stores/TagStore";
+import UserStore from "stores/UserStore";
+import VersionStore from "stores/VersionStore";
+import VolumeStore from "stores/VolumeStore";
+import AllocationSourceStore from "stores/AllocationSourceStore";
+
+// Create a dictionary of all stores
+//
+// We cannot rely on `import stores from 'stores'`, since it is populated too
+// late. This wrapper component wraps the module when it is defined and
+// stores.<STORE_NAME> isn't available then.
+let stores = {
+    AllocationStore, BadgeStore, ClientCredentialStore, ExternalLinkStore, HelpLinkStore, ImageStore, ImageVersionStore, ImageVersionMembershipStore, ImageVersionLicenseStore, ImageVersionScriptStore, IdentityStore, ImageBookmarkStore, InstanceHistoryStore, ImageRequestStore, InstanceStore, InstanceTagStore, LicenseStore, ScriptStore, MaintenanceMessageStore, MyBadgeStore, MembershipStore, ProfileStore, ProjectStore, ProjectExternalLinkStore, ProjectImageStore, ProjectInstanceStore, ProjectVolumeStore, ProviderMachineStore, ProviderStore, ResourceRequestStore, IdentityMembershipStore, StatusStore, SSHKeyStore, QuotaStore, SizeStore, TagStore, UserStore, VersionStore, VolumeStore, AllocationSourceStore,
+}
+
+// Add change listeners to the component for the following stores. The stores
+// will be added to the props of the instance via a prop called subscriptions.
+// Inside the wrapped component:
+//
+//     let { ProfileStore } = this.props.subscriptions;
+//
+// Only stores included in storeNames will be visible to the component
+export default function(component, storeNames) {
+    let subscriptions = {};
+    storeNames.forEach(name => {
+        if (stores[name]) {
+            subscriptions[name] = stores[name];
+        }
+    });
+
+    return React.createClass({
+        componentDidMount: function() {
+            Object.values(subscriptions)
+                  .forEach(store => {
+                      store.addChangeListener(this.updateState)
+                  });
+        },
+
+        componentWillUnmount: function() {
+            Object.values(subscriptions)
+                  .forEach(store => {
+                      store.removeChangeListener(this.updateState)
+                  });
+        },
+
+        updateState: function() {
+            this.forceUpdate();
+        },
+
+        render: function() {
+            // Define the props so that we extend props, with a new prop:
+            // 'subscriptions'
+            let newProps = Object.assign(
+                    {},
+                    this.props,
+                    { subscriptions });
+            return React.createElement(component, newProps);
```

Comment:
I want to say React.cloneElement is what we want here, since we have already created the element it is probably faster to clone it. At least that is what I've seen used in similar cases. [This](http://stackoverflow.com/a/35616144) seems to support that but I don't really know for sure just putting it out there.